### PR TITLE
Improve setup and seed

### PR DESCRIPTION
**Pull request by teadur**
Retry of #1296 
First part of fixing #960, install gem deps and setup db in bin/setup.
Install gem build deps in bin/setup.deps.
Provide db:seed with initial data.

New codeclimate issues are result of rubocop-i version increase not my code changes, because of that i think it makes more sense to merge the PR and work on the new issues gradually not in this PR.

### CODE REVIEW

File: bin/setup.deps [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+# path to your application root.
+APP_ROOT = Pathname.new File.expand_path('../../',  __FILE__)
+
+
+def system!(*args)
+  system(*args) || abort("\n== Command #{args} failed ==")
+end
+
+def get_uid()
```
Comment (artur-beljajev) [inline bin/setup.deps]: `effective_user_id` might explain the intention better. `()` can be skipped.

File: bin/setup.deps [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  uid = `id -u`
+  uid.delete!("\n")
```
Comment (artur-beljajev) [inline bin/setup.deps]: Perhaps https://ruby-doc.org/core-2.4.5/String.html#method-i-strip is a better fit?

File: bin/setup.deps [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  unless Dir.exist?("#{ENV['HOME']}/.rbenv/plugins/ruby-build/")
+    system! 'git clone https://github.com/sstephenson/ruby-build.git $HOME/.rbenv/plugins/ruby-build'
+    system! 'echo export PATH="$HOME/.rbenv/plugins/ruby-build/bin:$PATH" >> ~/.bashrc'
+  end
+  unless Dir.exist?("#{ENV['HOME']}/.rbenv/plugins/rbenv-default-gems/")
+    system! 'git clone https://github.com/rbenv/rbenv-default-gems.git $HOME/.rbenv/plugins/rbenv-default-gems'
+    unless File.exist?("#{ENV['HOME']}/.rbenv/default-gems")
+      system! 'echo "bundler" > ~/.rbenv/default-gems'
+      end
+  end
```
Comment (artur-beljajev) [inline bin/setup.deps]: Why this is needed? Isn't it just a matter of putting few command there?

File: bin/setup.deps [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+Dir.chdir APP_ROOT do
+  puts '== Installing application build deps =='
+  sudo_prefix = ''
+  sudo_prefix = "sudo" unless get_uid == '0'
+  system! "#{sudo_prefix} apt-get update && #{sudo_prefix} apt-get -y --no-install-recommends install libxml2 libxml2-dev postgresql-client postgresql-client-common libpq-dev"
+
+  puts "== Installing rbenv ruby manager to #{ENV['HOME']} =="
+  unless Dir.exist?("#{ENV['HOME']}/.rbenv/")
+    system! 'git clone https://github.com/sstephenson/rbenv.git $HOME/.rbenv'
```
Comment (artur-beljajev) [inline bin/setup.deps]: Perhaps `https://github.com/rbenv/rbenv.git` looks more official? This way you don't need to figure out who is that "sstephenson" every time you see this.

File: bin/setup.deps [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  unless Dir.exist?("#{ENV['HOME']}/.rbenv/plugins/rbenv-default-gems/")
+    system! 'git clone https://github.com/rbenv/rbenv-default-gems.git $HOME/.rbenv/plugins/rbenv-default-gems'
+    unless File.exist?("#{ENV['HOME']}/.rbenv/default-gems")
+      system! 'echo "bundler" > ~/.rbenv/default-gems'
+      end
+  end
+
+  # Include RBENV in path
+  ENV['PATH'] = ENV['HOME'] + "/.rbenv/bin:" + ENV['HOME'] + "/.rbenv/plugins/ruby-build/bin:" + ENV['PATH']
+  NEEDED_RUBY = `cat .ruby-version`.freeze
```
Comment (artur-beljajev) [inline bin/setup.deps]: `NEEDED_RUBY_VERSION` is more intention revealing.

File: bin/setup.deps [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+  if '!bundle check' == false
```
Comment (artur-beljajev) [inline bin/setup.deps]: `'!bundle check' == false` is always `false`. Did you mean backtick instead of a single quote?

File: db/seeds.rb [hunk 7]
```diff
@@ -1,2 +1,59 @@
...
+    username: 'admin',
+    email: 'admin@domain.tld',
+    password: 'adminadmin',
+    password_confirmation: 'adminadmin',
+    identity_code: '38001085718',
+    country_code: 'EE',
+    roles: ['admin']
+  )
+  # Required for creating registrar
+  Setting.where('var': 'registry_vat_prc').first_or_create(
```
Comment (artur-beljajev) [inline db/seeds.rb]: Duplicates https://github.com/internetee/registry/blob/master/config/app.yml#L59.

File: db/seeds.rb [hunk 8]
```diff
@@ -1,2 +1,59 @@
...
+    value: '0.2'
+  )
+  # First registrar
+  Registrar.where('name': 'Registrar First').first_or_create!(
```
Comment (artur-beljajev) [inline db/seeds.rb]: Why do you need to create a registrar? Given you create an `AdminUser` on line 5, he can login by itself and create registrars by himself. Unlikely anybody will benefit from such registrar.

File: db/seeds.rb [hunk 9]
```diff
@@ -1,2 +1,59 @@
...
+    address_country_code: 'EE',
+    address_state: 'Harjumaa',
+    address_city: 'Tallinn',
+    address_street: 'Tänav 1',
+    address_zip: '1234546',
+    email: 'registrar@first.tld',
+    code: 'REG1'
+  )
+
+#  registrar.accounts.create!(account_type: Account::CASH, currency: 'EUR')
```
Comment (artur-beljajev) [inline db/seeds.rb]: Remove it entirely if not needed. Git history has it anyway.

File: db/seeds.rb [hunk 10]
```diff
@@ -1,2 +1,59 @@
 # This file should contain all the record creation needed to seed the database with its default values.
 # The data can then be loaded with the rake db:seed (or created alongside the db with db:setup).
+# The data can then be loaded with the rake db:seed (or created alongside the db with db:setup).
+ActiveRecord::Base.transaction do
+  AdminUser.where('username': 'admin').first_or_create!(
```
Comment (artur-beljajev) [inline db/seeds.rb]: It should be `username: 'admin'` (`:username` is a symbol).

File: bin/setup.deps (hunk 3 above)
Comment (teadur) [reply bin/setup.deps]: Clone cannot happen to non empty directory so this logic is needed.

File: db/seeds.rb (hunk 7 above)
Comment (teadur) [reply db/seeds.rb]: I could not figure out how to load defaults variables from app.yml it isn't done automatically

File: db/seeds.rb (hunk 8 above)
Comment (teadur) [reply db/seeds.rb]: The idea is to have demo data set there to ease the testing of features in development

File: db/seeds.rb (hunk 9 above)
Comment (teadur) [reply db/seeds.rb]: They will be enabled in future commits

File: db/seeds.rb (hunk 8 above)
Comment (artur-beljajev) [reply db/seeds.rb]: `rake db:fixtures:load` should be used for this.

Comment (artur-beljajev) [reply db/seeds.rb]: Dummy data is already available in test/fixtures.

**Comment by artur-beljajev** [comment]
@teadur Wouldn't it be better to update Rubocop in a separate PR? It's kinda confusing why there are so many Rubocop issues.

**Comment by teadur** [comment]
@artur-beljajev rebased on master and will be squash merged to master.
@timo want todo it yourself or shall I ?

**Pull request merged**